examples/blog-api/src/main.rs

```diff
@@ -44,7 +43,7 @@ async fn main() -> Result<ExitCode, Error> {
     // Add a timeout middleware to the /api routes. This will prevent the
     // server from waiting indefinitely if we lose connection to the
     // database. For this example, we're using a 10 second timeout.
-    api.middleware(timeout(Duration::from_secs(10)));
+    api.middleware(Timeout::from_secs(10).or_service_unavailable());
```

Comment:
Constructor syntax for middleware makes the most sense. The only reasonable exception is the `Allow` middleware.

Preferring constructor syntax gives middleware authors the ability to write convenience methods such as `Timeout::from_secs`.